File: backend/src/main/java/com/back/team11/domain/member/repository/MemberRepository.java
package com.back.team11.domain.member.repository;

import com.back.team11.domain.member.entity.Member;
import com.back.team11.domain.member.entity.Provider;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member, Long> {

    Optional<Member> findByProviderAndProviderId(Provider provider, String providerId);

    Optional<Member> findByEmail(String email);

    boolean existsByEmail(String email);
}

File: backend/src/main/java/com/back/team11/domain/member/repository/MemberRepository.kt
package com.back.team11.domain.member.repository

import com.back.team11.domain.member.entity.Member
import com.back.team11.domain.member.entity.Provider
import org.springframework.data.jpa.repository.JpaRepository

interface MemberRepository : JpaRepository<Member, Long> {
    fun findByProviderAndProviderId(provider: Provider, providerId: String): Member?

    fun findByEmail(email: String): Member?

    fun findMemberById(id: Long): Member?

    fun existsByEmail(email: String): Boolean
}
